From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael <michael@debix.local>
Date: Thu, 27 Mar 2026 00:00:00 +0100
Subject: [PATCH] ASoC: codecs: add TI TAC5212 audio codec driver

Add Kconfig and Makefile entries for the TAC5212 codec driver.
The source files (tac5212.c, tac5212.h) are deployed separately.

Signed-off-by: Michael <michael@debix.local>
---
 sound/soc/codecs/Kconfig  | 7 +++++++
 sound/soc/codecs/Makefile | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/sound/soc/codecs/Kconfig b/sound/soc/codecs/Kconfig
--- a/sound/soc/codecs/Kconfig
+++ b/sound/soc/codecs/Kconfig
@@ -238,6 +238,7 @@
 	imply SND_SOC_STI_SAS
 	imply SND_SOC_TAS2552
 	imply SND_SOC_TAS2562
+	imply SND_SOC_TAC5212
 	imply SND_SOC_TAS2764
 	imply SND_SOC_TAS2770
 	imply SND_SOC_TAS2780
@@ -1760,6 +1761,12 @@
 config SND_SOC_STI_SAS
 	tristate "codec Audio support for STI SAS codec"

+config SND_SOC_TAC5212
+	tristate "Texas Instruments TAC5212 CODEC"
+	depends on I2C
+	select REGMAP_I2C
+	help
+	  Enable support for TI TAC5212 stereo audio ADC/DAC codec.

 config SND_SOC_TAS2552
 	tristate "Texas Instruments TAS2552 Mono Audio amplifier"
diff --git a/sound/soc/codecs/Makefile b/sound/soc/codecs/Makefile
--- a/sound/soc/codecs/Makefile
+++ b/sound/soc/codecs/Makefile
@@ -275,6 +275,7 @@
 snd-soc-tas5720-objs := tas5720.o
 snd-soc-tas5805m-objs := tas5805m.o
 snd-soc-tas6424-objs := tas6424.o
+snd-soc-tac5212-objs := tac5212.o
 snd-soc-tda7419-objs := tda7419.o
 snd-soc-tas2770-objs := tas2770.o
 snd-soc-tas2781-comlib-objs := tas2781-comlib.o
@@ -665,6 +666,7 @@
 obj-$(CONFIG_SND_SOC_TAS5720)	+= snd-soc-tas5720.o
 obj-$(CONFIG_SND_SOC_TAS5805M)	+= snd-soc-tas5805m.o
 obj-$(CONFIG_SND_SOC_TAS6424)	+= snd-soc-tas6424.o
+obj-$(CONFIG_SND_SOC_TAC5212)	+= snd-soc-tac5212.o
 obj-$(CONFIG_SND_SOC_TDA7419)	+= snd-soc-tda7419.o
 obj-$(CONFIG_SND_SOC_TAS2770) += snd-soc-tas2770.o
 obj-$(CONFIG_SND_SOC_TFA9879)	+= snd-soc-tfa9879.o
